### PR TITLE
module warnings: add a pref to hide them

### PR DESCRIPTION
**Pull request by aurelienpierre**
I'm an acrobat, I don't do safety nets. Also, it's tiresome to get nagged all the time about valid stuff that is indeed non-standard.

### CODE REVIEW

**Comment by jenshannoschwalm** [comment]
I agree, the red warning might be offensive and would lead to obstruction of the acrobat being an artist. But what about not removing the warning at all but leave a little dot or so as a reminder?

File: data/darktableconfig.xml.in [hunk 1]
```diff
@@ -2800,6 +2800,13 @@
     <shortdescription>detect monochrome previews</shortdescription>
     <longdescription>many monochrome images can be identified via exif and preview data. beware: this slows down imports and reading exif data</longdescription>
  </dtconfig>
+ <dtconfig prefs="processing">
+    <name>plugins/darkroom/show_warnings</name>
+    <type>bool</type>
+    <default>true</default>
+    <shortdescription>show warning messages</shortdescription>
+    <longdescription>display messages in modules to warn beginner users when non-standard and possibly harmful settings are used in the pipeline.\nthese messages can be false-positive and should be disregarded if you know what you are doing. This option will hide them all the time.</longdescription>
```
Comment (agusmba) [inline data/darktableconfig.xml.in:2808]: In longdescription: "These messages..." case after period.

Comment (elstoc) [reply data/darktableconfig.xml.in:2808]: Wow you're opening up a big debate there :)

Comment (TurboGit) [inline data/darktableconfig.xml.in:2808]: This -> this

File: src/develop/imageop.c [hunk 2]
```diff
@@ -3077,7 +3077,10 @@ void dt_iop_cancel_history_update(dt_iop_module_t *module)
 
 char *dt_iop_warning_message(const char *message)
 {
-  return g_strdup_printf("<span foreground='red'>⚠</span> %s", message);
+  if(dt_conf_get_bool("plugins/darkroom/show_warnings"))
```
Comment (TurboGit) [inline src/develop/imageop.c:3080]: This seems to only remove the "red" on the message. But it is still displayed or not ?

Comment (aurelienpierre) [reply src/develop/imageop.c:3080]: This part acts on the module title, despite what the function name says (I got confused too). We simply remove the warning sign here, then the module title shows as usual.

**Comment by aurelienpierre** [comment]
> But what about not removing the warning at all but leave a little dot or so as a reminder?

Meh. I'd like it as clean as possible. We managed without warnings for 12 years.

**Review by TurboGit — APPROVED** [review]
Ok, let's move on with this.

**Pull request merged**